Added SVG NFT metadata for token uri on stake NFT

keirongulrajani opened the pull request:
## Scope
Changes for [MP-267](https://madhive.atlassian.net/browse/MP-267)

- Added a token uri descriptor for the NFTs
- Added on chain svg generation for the stake positions

## Why?

Why are we doing this?

## Todos
If any, what are the follow-up tasks required other than merging this PR?  Have they been arranged?
- [ ] ???

nelsonhp commented:
Everything looks good except the merge conflict from other PRs. I have fixed the pygen.py file so that ImmutableAuth may be re-generated. Please regen this file and fix conflicts in setup.ts

keirongulrajani commented:
> Everything looks good except the merge conflict from other PRs. I have fixed the pygen.py file so that ImmutableAuth may be re-generated. Please regen this file and fix conflicts in setup.ts

Just rebased from the base branch and addressed the conflicts 👍

keirongulrajani commented:
...and thats really annoying that the bugs/code smells flagged by SonarCloud are ones taken from upstream i.e. nothing to do with my changes 🤦‍♂️
